lib/uplink/ext/test_utils.go

```diff
@@ -0,0 +1,155 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package main
+
+// #cgo CFLAGS: -g -Wall
+// #include <stdlib.h>
+// #ifndef STORJ_HEADERS
+//   #define STORJ_HEADERS
+//   #include "c/headers/main.h"
+// #endif
+import "C"
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"testing"
+
+	"github.com/skyrings/skyring-common/tools/uuid"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+
+	"storj.io/storj/internal/testcontext"
+	"storj.io/storj/internal/testplanet"
+	"storj.io/storj/lib/uplink"
+	"storj.io/storj/pkg/macaroon"
+	"storj.io/storj/pkg/storj"
+	"storj.io/storj/satellite/console"
+)
+
+// C types
+type Cchar = *C.char
+
+// Ref types
+type CAPIKeyRef = C.APIKeyRef_t
+type CUplinkRef = C.UplinkRef_t
+
+var (
+	cLibDir, cSrcDir, cTestsDir, libuplink string
+
+	testConfig = new(uplink.Config)
+	ciphers    = []storj.CipherSuite{storj.EncNull, storj.EncAESGCM, storj.EncSecretBox}
+)
+
+func init() {
+	// TODO: is there a cleaner way to do this?
+	_, thisFile, _, _ := runtime.Caller(0)
+	cLibDir = filepath.Join(filepath.Dir(thisFile), "c")
+	cSrcDir = filepath.Join(cLibDir, "src")
+	cTestsDir = filepath.Join(cLibDir, "tests")
+	libuplink = filepath.Join(cLibDir, "..", "uplink-cgo.so")
+
+	testConfig.Volatile.TLS.SkipPeerCAWhitelist = true
+}
+
+func runCTests(t *testing.T, ctx *testcontext.Context, envVars []string, srcGlobs ...string) {
+	srcGlobs = append([]string{
+		libuplink,
+		filepath.Join(cTestsDir, "unity.c"),
+		filepath.Join(cTestsDir, "helpers.c"),
+		filepath.Join(cSrcDir, "*.c"),
+	}, srcGlobs...)
+	testBinPath := ctx.CompileC(srcGlobs...)
+	commandPath := testBinPath
+
+	if dir, ok := os.LookupEnv("STORJ_DEBUG"); ok {
```

Comment:
please add comment what's happening here

Reply:
Sure; this is just for debugging. Because we're using a test context for all the bin paths (and we're deleting them after the tests finish), if you need to debug a test binary (with gdb, for example), this lets you copy those bins to the path in the `STORJ_DEBUG` env var.

If it makes sense to keep this, that env var probably needs a better name. Thoughts?

Reply:
hmm I don't know if we should keep this. If you used it at least once then most probably yes. Regarding name, a better name would be nice. Maybe something with `STORJ_TEST_BINARY_OUTPUT` or `STORJ_TEST_BINARY_COPY` but I don't have a strong opinion. If you decide to leave it in the code add a comment to code why its there and how it can be used.